test/unit-test/mqtt_agent_utest.c

```diff
@@ -701,12 +707,93 @@ void test_MQTTAgent_Unsubscribe_success( void )
 
 /* ========================================================================== */
 
+/**
+ * @brief Test MQTTAgent_Publish() with invalid parameters.
+ */
+void test_MQTTAgent_Publish_Invalid_Parameters( void )
+{
+    MQTTAgentContext_t agentContext = { 0 };
+    MQTTStatus_t mqttStatus;
+    CommandInfo_t commandInfo = { 0 };
+    Command_t command = { 0 };
+    MQTTPublishInfo_t publishInfo = { 0 };
+
+    setupAgentContext( &agentContext );
+    pCommandToReturn = &command;
+    commandInfo.cmdCompleteCallback = stubCompletionCallback;
+    /* Test topic name. */
+    publishInfo.pTopicName = "test";
+    publishInfo.topicNameLength = 4;
+
+    /* NULL parameters. */
+    mqttStatus = MQTTAgent_Publish( NULL, &publishInfo, &commandInfo );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    mqttStatus = MQTTAgent_Publish( &agentContext, NULL, &commandInfo );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    mqttStatus = MQTTAgent_Publish( &agentContext, &publishInfo, NULL );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    /* This needs to be large enough to hold the PUBLISH:
+     * 1 byte: control header
+     * 1 byte: remaining length
+     * 2 bytes: topic name length
+     * 1+ bytes: topic name.
+     * For this test case, the buffer must have size at least
+     * 1+1+2+4=8. */
+    agentContext.mqttContext.networkBuffer.size = 6;
+    mqttStatus = MQTTAgent_Publish( &agentContext, &publishInfo, &commandInfo );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+}
+
+/**
+ * @brief Test that a bad parameter is returned when there
```

Comment:
I see MQTTNoMemory in the assertion.  Does the comment need to change?

Reply:
Updated the comment